### PR TITLE
Remove WP Stories feature config and the unused code

### LINKED ISSUE #14318: Remove WP Stories feature flag
The task is to remove `WPStoriesFeatureConfig` and the now unused code.

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #14318 

This PR removes the WPStoriesEnabled flag because it was already released to all the users and it will not be reverted. The code should be quite straightforward.

To test:
- Check you can correctly add stories on a WPCom site
- Check the stories are disabled on a non-Jetpack self-hosted site

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt [hunk 1]
```diff
@@ -321,7 +319,7 @@ class WPMainActivityViewModel @Inject constructor(
     }
 
     private fun shouldShowStories(site: SiteModel?): Boolean {
-        return wpStoriesFeatureConfig.isEnabled() && SiteUtils.supportsStoriesFeature(site)
+        return SiteUtils.supportsStoriesFeature(site)
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt]: What do you think about removing this simple method and replacing wherever it's called with `SiteUtils.supportsStoriesFeature(site)` directly?

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt]: We can leave it if you think it adds clarity - it's just that the occurrences where this is used are already giving reading clarity by assigning the return value to a descriptive named val i.e. `val shouldShowStories = shouldShowStories(site)`

Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt]: I've implemented your suggestion in a9b2936421, I think it makes sense 👍

**Comment by planarvoid** [comment]
I've implemented your suggestion so this should be ready for another check @mzorz 👍

**Review by mzorz — APPROVED** [review]
Tested as per the instructions and worked as expected 👍 
LGTM :shipit:

**Pull request merged**